src/components/Footer.js

```diff
@@ -15,18 +27,26 @@ export default function Footer() {
             <NavLink to="/">
               <button className="Footer__btn Footer__btn--left">Back</button>
             </NavLink>
-            <NavLink to="/sudoku">
-              <button className="Footer__btn Footer__btn--right">Enter</button>
+            <NavLink to={isRoomSelected ? "/sudoku" : "/map"}>
+              <button
+                className="Footer__btn Footer__btn--right"
+                onClick={handleEnterClick}
+              >
+                Enter
+              </button>
```

Comment:
You could just write: 

onClick={onEnterClick}

Also: I think it is strongly discouraged to use buttons inside a tags since they are prone to bubble. 
From what I can see you could just remove the button tag and give the onSubmit, onClick to the a tag or remove the a tag and have the button do a history.push onClick

Reply:
Oh ok, I wasn't aware of that. Thanks for the hint! I will change it. 
Using history.push sounds like the good option.